libs/@local/hash-isomorphic-utils/src/system-types/linear/attachment.ts

```diff
@@ -310,37 +490,65 @@ export type BelongsToIssueProperties1 = LinkProperties;
 
 export type BelongsToIssueProperties2 = {};
 
+export type BelongsToIssuePropertiesWithMetadata = {
+  metadata?: ObjectMetadata;
+  value: {};
+};
+
 /**
  * Indicates if attachments for the same source application should be grouped in the Linear UI.
  */
 export type GroupBySourcePropertyValue = BooleanDataType;
 
+export type GroupBySourcePropertyValueWithMetadata =
+  BooleanDataTypeWithMetadata;
+
 /**
  * Custom metadata related to the attachment.
  */
 export type MetadataPropertyValue = ObjectDataType;
 
+export type MetadataPropertyValueWithMetadata = ObjectDataTypeWithMetadata;
+
 /**
  * An opaque, untyped JSON object
  */
 export type ObjectDataType = {};
 
+export type ObjectDataTypeWithMetadata = {
+  value: ObjectDataType;
+  metadata: ObjectDataTypeMetadata;
+};
+export type ObjectDataTypeMetadata = {
+  provenance?: PropertyProvenance;
+  confidence?: Confidence;
+  dataTypeId: "https://blockprotocol.org/@blockprotocol/types/data-type/object/v/1";
+};
```

Comment:
Should this definition live in this file?

Reply:
How the codegen file allocation is defined is that if a type is used by more than one file, it is moved into a `shared.ts`. Within the scope of the `linear` types, `ObjectDataType` is only used by `attachment`, which is why it stays here.

We can change that logic if we want, e.g. _always_ move data types into a `shared.ts` file, although it might be weird for very specific data types that clearly 'belong' to some entity type.

Reply:
We could also update the codegen so that we pass all the different namespace types at once, rather than as separate calls to `codegen`, in which case it's more likely that things get used by multiple entity types.

Reply:
> Within the scope of the linear types, ObjectDataType is only used by attachment, which is why it stays here.

I think it's fine to leave it as it is then